### PR TITLE
Digital Guardian DLP Data Connector

### PR DESCRIPTION
**Pull request by vu-socprime**
## Proposed Changes

  - add Digital Guardian DLP Data Connector

### CODE REVIEW

**Comment by v-jayakal** [comment]
@oshezaf , can you please review the parser? Thanks.

**Review by v-admahe — CHANGES_REQUESTED** [review]

File: Solutions/DigitalGuardianDLP/Parsers/DigitalGuardianDLPEvent.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// Usage Instructions: 
+// Paste below query in log analytics, click on Save button and select as Function from drop down by specifying function name as DigitalGuardianDLPEvent and specifying Legacy Category.
+// This function maps Digital Guargian DLP events to normalization schema (https://docs.microsoft.com/azure/sentinel/normalization-schema).
+// Function usually takes 10-15 minutes to activate. You can then use function alias from any other queries (e.g. DigitalGuardianDLPEvent | take 10).
+// Reference : Using functions in Azure monitor log queries : https://docs.microsoft.com/azure/azure-monitor/log-query/functions
+Syslog
+| where SyslogMessage contains 'managed_device_id' and SyslogMessage contains 'number_of_incidents'
+| mv-apply ExtractedFields = extract_all(@'\s(?P<key>[a-zA-Z0-9-_]+)=\"?(?P<value>[a-zA-Z0-9-_:/@.,#{}>< ]+)\"?', dynamic(["key","value"]), SyslogMessage) on (
+    project packed = pack(tostring(ExtractedFields[0]), tostring(ExtractedFields[1]))
+    | summarize bag = make_bag(p)
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/DigitalGuardianDLP/Parsers/DigitalGuardianDLPEvent.txt]: fixed

File: Solutions/DigitalGuardianDLP/Data Connectors/Connector_DigitalGuardian_Syslog.json [hunk 2]
```diff
@@ -0,0 +1,153 @@
+{
+    "id": "DigitalGuardianDLP",
+    "title": "Digital Guardian Data Loss Prevention",
+    "publisher": "Digital Guardian",
+    "descriptionMarkdown": "[Digital Guardian Data Loss Prevention (DLP)](https://digitalguardian.com/platform-overview) data connector provides the capability to ingest Digital Guardian DLP logs into Azure Sentinel.",
+    "additionalRequirementBanner": "This data connector depends on a parser based on a Kusto Function to work as expected [**DigitalGuardianDLPEvent**](https://aka.ms/sentinel-DigitalGuardianDLP-parser) which is deployed with the Azure Sentinel Solution.",
+    "graphQueries": [
+        {
+            "metricName": "Total data received",
+            "legend": "Syslog (DigitalGuardianDLPEvent)",
```
[the remark this answers is not in the document]
Comment (vu-socprime) [reply Solutions/DigitalGuardianDLP/Data Connectors/Connector_DigitalGuardian_Syslog.json]: updated

**Comment by v-admahe** [comment]
@vu-socprime : We are getting error in parser , it didnot find the timestamp field.

image

We have check that this field is available in syslog message but at the time we make a pack of two syslog message field this field timestamp field is not included.

Thanks,
Aditi Maheshwari

**Comment by v-admahe** [comment]
@shainw could you please review the parser.

Thanks,
Aditi

**Comment by vu-socprime** [comment]
> @vu-socprime : We are getting error in parser , it didnot find the timestamp field.
> 
> image
> 
> We have check that this field is available in syslog message but at the time we make a pack of two syslog message field this field timestamp field is not included.
> 
> Thanks,
> Aditi Maheshwari

@v-admahe This error occurs when there are no data in the table. Try to use other time period in the query that contains DigitalGuardian events in Syslog table.

**Review by v-rucdu — APPROVED** [review]

**Review by v-jayakal — APPROVED** [review]

**Pull request merged**